backend/src/schema/resolvers/moderation.spec.js

```diff
@@ -54,369 +63,369 @@ describe('disable', () => {
-          const before = { Post: [{ id: 'p9', disabled: false }] }
-          const expected = { Post: [{ id: 'p9', disabled: true }] }
-
-          await setup()
-          await expect(client.request('{ Post { id disabled } }')).resolves.toEqual(before)
-          await action()
-          await expect(client.request('{ Post(disabled: true) { id disabled } }')).resolves.toEqual(
-            expected,
-          )
         })
+        authenticatedUser = await nonModerator.toJson()
       })
-    })
-  })
-})
-
-describe('enable', () => {
-  const mutation = gql`
-    mutation($id: ID!) {
-      enable(id: $id)
-    }
-  `
-  let variables
-
-  const action = async () => {
-    return client.request(mutation, variables)
-  }
-
-  beforeEach(() => {
-    // our defaul set of variables
-    variables = {
-      id: 'blabla',
-    }
-  })
-
-  it('throws authorization error', async () => {
-    await setup()
-    await expect(action()).rejects.toThrow('Not Authorised')
-  })
-
-  describe('authenticated', () => {
-    beforeEach(() => {
-      authenticateClient = setupAuthenticateClient({
-        email: 'user@example.org',
-        password: '1234',
+      it('throws authorization error', async () => {
+        const { errors } = await mutate({ mutation, variables })
+        expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
       })
     })
 
-    it('throws authorization error', async () => {
-      await setup()
-      await expect(action()).rejects.toThrow('Not Authorised')
-    })
-
-    describe('as moderator', () => {
+    describe('I am a moderator', () => {
+      let moderator
       beforeEach(async () => {
-        authenticateClient = setupAuthenticateClient({
-          role: 'moderator',
-          email: 'someuser@example.org',
+        moderator = await factory.create('User', {
+          id: 'moderator-id',
+          name: 'Moderator',
+          email: 'moderator@example.org',
           password: '1234',
+          role: 'moderator',
         })
+        authenticatedUser = await moderator.toJson()
       })
 
-      describe('on something that is not a (Comment|Post|User) ', () => {
+      describe('moderate a resource that is not a (Comment|Post|User) ', () => {
         beforeEach(async () => {
           variables = {
-            id: 't23',
-          }
-          createResource = () => {
-            // we cannot create a :DISABLED relationship here
-            return Promise.all([factory.create('Tag', { id: 't23' })])
+            id: 'sample-tag-id',
           }
+          await Promise.all([factory.create('Tag', { id: 'sample-tag-id' })])
         })
 
         it('returns null', async () => {
-          const expected = { enable: null }
-          await setup()
-          await expect(action()).resolves.toEqual(expected)
+          const expected = { data: { disable: null } }
+          await expect(mutate({ mutation, variables })).resolves.toMatchObject(expected)
```

Comment:
Not too much code here, so you could merge these two lines,

Reply:
Done